colossus/src/main/scala/colossus/protocols/http/HttpRequestParser.scala

```diff
@@ -11,18 +11,20 @@ object HttpRequestParser {
   
   def apply() = httpRequest
 
-  protected def httpRequest: Parser[HttpRequest] = httpHead |> {head =>
+  protected def httpRequest: Parser[HttpRequest] = httpHead |> { head =>
+    val contentType = head.headers.firstValue("content-type")
```

Comment:
So this is not the right place to do this logic.  `headers.firstValue` ends up scanning the list of headers and is not efficient.  Instead, you want to look for this header during the parsing itself, which uses some specialized tools to do this as efficiently as possible.  See [HeadersBuilder](https://github.com/tumblr/colossus/blob/3a296309c97adf6bf878f69dc2c82c6d6dcfaa61/colossus/src/main/scala/colossus/protocols/http/HttpParse.scala#L28) and [ParsedHttpHeaders](https://github.com/tumblr/colossus/blob/3a296309c97adf6bf878f69dc2c82c6d6dcfaa61/colossus/src/main/scala/colossus/protocols/http/Header.scala#L184).  It should be pretty clear how to add the extra case for content-type, since it should work exactly the same as the other headers, but don't hesitate to ask if you have questions.

This applies to parsing both requests and responses.